src/screens/SiteDetailMapViewScreen.js

```diff
@@ -0,0 +1,108 @@
+import React, {Component} from 'react';
+import {View, StyleSheet, Linking, Platform} from 'react-native';
+import {Card, Text} from 'react-native-elements';
+import {Icon} from 'react-native-elements';
+import {connect} from 'react-redux';
+import {MapView} from 'expo';
+
+const {Marker} = MapView;
+
+import _ from 'lodash';
+
+import { campsite, common} from '../locale.en';
+import {campsiteIcon, linkColorBlue} from "../styles";
+
+const {click_for_maps} = campsite;
+
+class SiteMapViewScreen extends Component {
+
+    componentWillMount() {
+        const {selectedSite, navigation: {setParams}} = this.props;
+
+        setParams({siteTitle: selectedSite.title});
+    }
+
+    setMarkerRef = (ref) => {
+        this.marker = ref
+    };
+
+    static navigationOptions = (props) => {
+        const {navigation: {state: {params = {}}}} = props;
+        const {siteTitle} = params;
+
+        return {
+            headerTitle: _.isUndefined(siteTitle) ? '' : siteTitle
+        }
+
+    };
+
+    onClickMarker = () => {
+        const {selectedSite} = this.props;
+        const {coordinate} = selectedSite;
+        const zoomLevel = 15;
+        const mapLink = `http://maps.google.com/maps?q=${coordinate.latitude},${coordinate.longitude}&z=${zoomLevel}`;
+
+        Linking.openURL(mapLink).catch(err => console.error('An error occurred', err));
+    };
+
+
+    renderSiteDetailMapViewScreen = () => {
+        const {fillScreen} = styles;
+        const {selectedSite} = this.props;
+        const {coordinate} = selectedSite;
+
+        if (selectedSite) {
+            return (
+
+                <MapView
+                    style={fillScreen}
+                    onRegionChangeComplete={() => this.marker.showCallout()}
+                    initialRegion={{
+                        longitude: coordinate.longitude,
+                        latitude: coordinate.latitude,
+                        longitudeDelta: 1,
+                        latitudeDelta: 1
+                    }}
+                >
+                    <Marker
+                        ref={this.setMarkerRef}
+                        coordinate={coordinate}
+                        showCallout={true}
+                        title={click_for_maps}
+                        onCalloutPress={this.onClickMarker}
+                        onPress={this.onClickMarker}
```

Comment:
If the user clicks on either the callout or the marker itself, open Google Maps. Also, the title of the marker is the `click_for_maps` helper text that lets them know that it'll open Google Maps.